src/service/resources-impl.js

```diff
@@ -336,20 +336,30 @@ export class Resources {
       element.id = 'AMP_' + resource.getId();
     }
     this.resources_.push(resource);
+    this.buildOrScheduleBuildForElement_(element);
+    dev().fine(TAG_, 'element added:', resource.debugid);
+  }
 
+  /**
+   * Builds the element if ready to be built, otherwise adds it to pending resources.
+   * @private
+   */
+  buildOrScheduleBuildForElement_(element) {
+    const resource = this.getResourceForElement(element);
     if (this.isRuntimeOn_) {
+      if (element.isBuilt()) {
```

Comment:
One thing, however. You are loosing `schedulePass` on `add` in this case. Intended? Is there a reason why you wanted to change the old behavior?